### PR TITLE
Fix include ordering to give self-contained units

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Preparation for move to clang format 5. Fixes compilation problems that arise due to include file reordering.

This has exposed an issue with the [NeXus](https://github.com/nexusformat) library where its headers cannot be included in an arbitrary order. This has been reported at nexusformat/code#457.

**To test:**

Code review.

Fixes #xxxx. 

Does this update require release notes?
- [ ] Yes
- [X] No

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/inc/MantidDataHandling/LoadEventNexus.h [hunk 1]
```diff
@@ -17,8 +17,11 @@
 #undef CHAR
 #endif
 
-#include <nexus/NeXusException.hpp>
+// clang-format off
 #include <nexus/NeXusFile.hpp>
+#include <nexus/NeXusException.hpp>
+// clang-format on
+#include <nexus/NeXusException.hpp>
```
Comment (AndreiSavici) [inline Framework/DataHandling/inc/MantidDataHandling/LoadEventNexus.h]: Why include this twice?

File: Framework/DataHandling/src/LoadMcStas.cpp [hunk 2]
```diff
@@ -14,8 +14,11 @@
 #include "MantidGeometry/Instrument/InstrumentDefinitionParser.h"
 
 #include <boost/algorithm/string.hpp>
-#include <nexus/NeXusException.hpp>
+// clang-format off
 #include <nexus/NeXusFile.hpp>
+#include <nexus/NeXusException.hpp>
+// clang-format on
+#include <nexus/NeXusException.hpp>
```
Comment (AndreiSavici) [inline Framework/DataHandling/src/LoadMcStas.cpp]: Same as before

File: Framework/DataObjects/src/PeaksWorkspace.cpp [hunk 3]
```diff
@@ -26,8 +26,11 @@
 #include <cstdlib>
 #include <exception>
 #include <fstream>
-#include <nexus/NeXusException.hpp>
+// clang-format off
 #include <nexus/NeXusFile.hpp>
+#include <nexus/NeXusException.hpp>
+// clang-format on
+#include <nexus/NeXusException.hpp>
 #include <ostream>
```
Comment (AndreiSavici) [inline Framework/DataObjects/src/PeaksWorkspace.cpp]: +1

**Review by AndreiSavici — CHANGES_REQUESTED** [review]

File: Framework/DataHandling/inc/MantidDataHandling/LoadEventNexus.h (hunk 1 above)
Comment (martyngigg) [reply Framework/DataHandling/inc/MantidDataHandling/LoadEventNexus.h]: Nice catch. I did this with a script and didn't check all of the files by hand.

**Comment by martyngigg** [comment]
Thanks for pointing those out, Andrei. I've fixed them now.

**Review by AndreiSavici — APPROVED** [review]

**Pull request merged**